### PR TITLE
Document schema with diagrams

### PR DESCRIPTION
**Pull request by ipatrol**
Ok, I think I actually got the revision history clean this time.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,643 @@
+# If this isn't enough to cover all your frameworks, editors, and IDEs, then I have no idea what you are doing
```
Comment (shm0o) [inline .gitignore]: You already have been told in https://github.com/nyaadevs/nyaa/pull/154#issuecomment-303216953 that most of these don't apply to this project or are better suited for the developer's global configuration.

**Comment by aldacron** [comment]
wait, what is the goal of this?

**Comment by ipatrol** [comment]
Would you rather it be revised to include only software that is actually used by the project?

**Comment by TheAMM** [comment]
@ipatrol The point is not to touch the `.git*` files at all.
Personally I don't think we need this either.

**Comment by FichteFoll** [comment]
It's nice to have something to look at when you're not involved with the project but intend to, or are just a little interested, but it needs to be updated whenever the schema changes as well, as it would otherwise be misleading.

**Comment by ipatrol** [comment]
Ok. Now only thing changed in the .gitignore is the addition of the virtualenv stuff, since the project instructions say to use it.

**Comment by Arylide** [comment]
Seemingly abandoned PR, closing for now. If anyone wants to work on this please submit a new PR and reference this one.

**Pull request closed without merging**